scipy/stats/morestats.py

```diff
@@ -1122,7 +1129,14 @@ def _eval_pearsonr(lmbda, xvals, samps):
             # correlation.
             y = boxcox(samps, lmbda)
             yvals = np.sort(y)
-            r, prob = stats.pearsonr(xvals, yvals)
+            # When all of yvals are equal, `stats.pearsonr` fails due to a
+            # division by zero error; since we want to approximate a normal
+            # distribution anyway we just discard these yvals and say that
+            # the correlation is zero
+            if yvals.max() - yvals.min() < 1E-21:
```

Comment:
Changed the arbitrary precision check to `np.unique`; to my surprise it seems to work just fine. I wouldn't be surprised if numpy is just doing something similar behind the scenes.

`np.nan` should not be returned here because it messes with the optimizer (and tests appear fail). I have not tested what the behavior would be when combined with gh-9562 but I would recommend against merging this change and that one simultaneously since they may affect each other. Although the tests I wrote for this change should be sufficient to tell whether a merge with gh-9562.